src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -299,6 +295,14 @@ export default class SurvivalChart
         this.tooltipMouseLeave = this.tooltipMouseLeave.bind(this);
     }
 
+    componentWillReceiveProps(nextProps: Readonly<ISurvivalChartProps>) {
+        // when click a new chart, reset the slider
+        if (this.currentChartId !== nextProps.chartId) {
```

Comment:
i would expect that chartId would be an immutable property of a component that creates the chart.   Unless I misunderstand, we should remount the component when the id changes.  That should also have the effect of resetting UI.   You can achieve this by giving the <SurvivalChart> a key attribute that depends on chartId.